### PR TITLE
Move button and context menu code into bootstrap, inline icon svg

### PR DESCRIPTION
**Pull request by 6a68**

### CODE REVIEW

File: addon/bootstrap.js [hunk 1]
```diff
@@ -51,11 +57,21 @@ function startup(data, reason) { // eslint-disable-line no-unused-vars
   prefObserver.register();
   addonResourceURI = data.resourceURI;
   // eslint-disable-next-line promise/catch-or-return
-  appStartupPromise.then(handleStartup);
+  appStartupPromise.then(initUI);
+}
+
+function initUI() {
+  initButton();
+  // TODO: check the pref state, then either init or shutdown the context menu
```
Comment (6a68) [inline addon/bootstrap.js]: added to todo list in #2877

File: addon/bootstrap.js [hunk 2]
```diff
@@ -77,7 +93,21 @@ function shouldDisable() {
...
 
+// the postmessage channel opened once the webextension is started
+let webExtensionPort;
+function onConnect(port) {
+  webExtensionPort = port;
+}
+
+function onClick() {
+  if (!webExtensionPort) { return; } // just do nothing till the port is ready
+  // TODO: figure out how to send the tab ID over
```
Comment (6a68) [inline addon/bootstrap.js]: added to todo list in #2877

File: addon/bootstrap.js [hunk 3]
```diff
@@ -77,7 +93,21 @@ function shouldDisable() {
...
+  webExtensionPort.postMessage({ content: "click" });
+}
+
+
 function handleStartup() {
+  // TODO: check the pref _before_ calling into ExtensionsUtils.
```
Comment (6a68) [inline addon/bootstrap.js]: this is an older todo, so it might make sense to fix directly in master. filed #2879

File: addon/bootstrap.js [hunk 4]
```diff
@@ -130,3 +164,68 @@ function handleMessage(msg, sender, sendReply) {
     return true;
   }
 }
+
+let buttonStyleSheet;
+
+// TODO: deal with toggling the prefs on/off
```
Comment (6a68) [inline addon/bootstrap.js]: duplicate todo, I guess. tracked in #2877

File: addon/bootstrap.js [hunk 5]
```diff
@@ -130,3 +164,68 @@ function handleMessage(msg, sender, sendReply) {
...
+  },
+  observe: function(aSubject, aTopic, aData) {
+    const document = aSubject.wrappedJSObject.menu.ownerDocument;
+    let item = document.getElementById("context-screenshots");
+    if (!item) {
+      item = document.createElement("menuitem");
+      item.setAttribute("id", "context-screenshots");
+      item.setAttribute("label", "screenshots"); // TODO solve l10n without webextensions :-(
+      item.setAttribute("oncommand", "window.alert('clicked screenshots context menu')");
+      // TODO: add an "accesskey" attribute?
```
Comment (6a68) [inline addon/bootstrap.js]: added to #2877

Comment (ianb) [inline addon/bootstrap.js]: Would that be a hotkey like asked for in https://bugzilla.mozilla.org/show_bug.cgi?id=1365757 ? Or just a key specific to the context menu?

File: addon/webextension/background/main.js [hunk 6]
```diff
@@ -295,5 +179,17 @@ this.main = (function() {
     return startSelectionWithOnboarding(sender.tab);
   });
 
+  // Open a port to let the embedding legacy addon tell us when a button is pressed
+  // TODO: do we need to specify the extensionID in the connect call?
```
Comment (6a68) [inline addon/webextension/background/main.js]: stale todo - the console.log is fired when the button is clicked repeatedly, so this code does work

File: addon/webextension/background/main.js [hunk 7]
```diff
@@ -84,113 +69,12 @@ this.main = (function() {
     });
   }
 
-  function shouldOpenMyShots(url) {
-    return /^about:(?:newtab|blank)/i.test(url) || /^resource:\/\/activity-streams\//i.test(url);
-  }
-
-  browser.browserAction.onClicked.addListener(catcher.watchFunction((tab) => {
```
Comment (ianb) [inline addon/webextension/background/main.js]: In lazy-background I changed this to a function, `main.onClicked = ...` – then I just changed how it was invoked. I think the same would be appropriate here.

File: addon/webextension/background/main.js [hunk 8]
```diff
@@ -295,5 +179,17 @@ this.main = (function() {
...
 
+  // Open a port to let the embedding legacy addon tell us when a button is pressed
+  // TODO: do we need to specify the extensionID in the connect call?
+  // TODO: as soon as the port exists, send a "hello" message over, so the embedding
+  // addon can get a reference to the port.
+  const port = browser.runtime.connect({name: "screenshots-legacy-connection"});
+  port.onMessage.addListener((message) => {
+    if (message.content === "click") {
+      console.log("got a click signal from the legacy addon");
+      // TODO: figure out how to get the tabID of the clicked tab
```
Comment (ianb) [inline addon/webextension/background/main.js]: Something like `browser.tabs.query({active: true}).then(...)`

File: addon/webextension/background/main.js [hunk 9]
```diff
@@ -84,113 +69,12 @@ this.main = (function() {
...
-    contexts: ["page"],
-    documentUrlPatterns: ["<all_urls>"]
-  }, () => {
-    // Note: unlike most browser.* functions this one does not return a promise
-    if (browser.runtime.lastError) {
-      catcher.unhandled(new Error(browser.runtime.lastError.message));
-    }
-  });
-
-  browser.contextMenus.onClicked.addListener(catcher.watchFunction((info, tab) => {
```
Comment (ianb) [inline addon/webextension/background/main.js]: This also could become `main.onClickedContextMenu`

**Comment by 6a68** [comment]
The existing changes in the lazybutton branch were rebased into the one commit in this branch, so that's why git is showing conflicts. I can just force-push this branch when it's good to land, and no code should be lost (though existing branches will be orphaned)

File: addon/bootstrap.js (hunk 5 above)
Comment (6a68) [inline addon/bootstrap.js]: I noticed an 'accesskey' attribute in the pocket code i borrowed from. Looking at MDN, it seems this attribute just underlines a specific character in the string: https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XUL/Attribute/accesskey

**Comment by 6a68** [comment]
Yeah, I can bring back the deleted code. For some reason, commenting it out with multiline comments was causing the build to break, which was the original motivation to delete the handlers.

**Comment by 6a68** [comment]
looks like we are not going in this direction after all 😅

**Pull request closed without merging**